airflow-core/src/airflow/utils/log/file_task_handler.py

```diff
@@ -647,7 +647,7 @@ def _read(
         if ti.state in (TaskInstanceState.RUNNING, TaskInstanceState.DEFERRED) and not has_k8s_exec_pod:
             sources, served_logs = self._read_from_logs_server(ti, worker_log_rel_path)
             source_list.extend(sources)
-        elif ti.state not in State.unfinished and not (local_logs or remote_logs):
+        elif (ti.state not in State.unfinished or ti.state == TaskInstanceState.UP_FOR_RETRY) and not (local_logs or remote_logs):
```

Comment:
```suggestion
elif (ti.state not in State.unfinished or ti.state == TaskInstanceState.UP_FOR_RETRY) and not (
    local_logs or remote_logs
):
```

To fix static checks